src/main/java/dev/morling/onebrc/CalculateAverage_justplainlaake.java

```diff
@@ -0,0 +1,439 @@
+                        s1.sum += s.sum;
+                        return s1;
+                    });
+                });
+            }
+            catch (InterruptedException | ExecutionException e1) {
+                e1.printStackTrace();
+            }
+        }
+
+        // Mark threadpool to be shutdown, call it here to let the threadpool finish out while the rest of the processing occurs
+        e.shutdown();
+
+        // Ordering and printing takes 50ms
+        Station[] nameOrdered = merged.toArray();// Turn the merged map into an array to quickly sort it
+
+        Arrays.sort(nameOrdered, (n1, n2) -> n1.name.compareTo(n2.name));// Sort based on name, this might be optimizable based on the longs of the name, but would likely only gain some ms??
+
+        // Print results to the sys out
+        System.out.print("{");
+        for (int i = 0; i < nameOrdered.length; i++) {
+            if (i != 0) {
+                System.out.print(", ");
+            }
+            System.out.print(nameOrdered[i]);
+        }
+        System.out.print("}\n");// Need newline character to meet specs
+    }
+
+    // Core processing functionality, processes a chunk of memory
+    private static OpenMap process(long fromAddress, long toAddress) {
+
+        OpenMap stationsLookup = new OpenMap();// Create a new map for this specific chunk, this is also the returned value for the callable
+
+        long blockStart = fromAddress;
+        long currentAddress = fromAddress;
+
+        while (currentAddress < toAddress) {// Just keep looping until we exhaust the chunk
+
+            long read = 0l;
+            short offset = -1;
+            // The hash is a long hash based on the murmur3 algorithm. Look at the getMurmurHash3 method to find link
+            long hash = 1;
+
+            while ((offset = getMaskOffset(read = UNSAFE.getLong(currentAddress), SEPERATOR_BYTE)) == -1) {// Read and compute the hash until we locate the seperator byte 59 or ';'
+                currentAddress += 8;// forwardscan
+                hash = (997 * hash) ^ (991 * getMurmurHash3(read));
+            }
+
+            // Compute the final hash based using the last read long but only the effective bits (anything before the byte 59 or ';').
+            // Using the OFFSET_CLEARS masks that are defined statically we can essentially segregate the important bits of the name based on the offset read above
+            hash = (997 * hash) ^ (991 * getMurmurHash3(read & OFFSET_CLEARS[offset]));
+
+            // Advance the current address/pointer to be 1 character past the end of the name Example: BillyJoel;29 would make the current address start at the '2' character
+            currentAddress += offset + 1;
+
+            Station station = stationsLookup.getOrCreate(hash, currentAddress, blockStart);
+            // Bet on the likelyhood that there are no collisions in the hashed name (Extremely Rare, more likely to get eaten by a shark in Kansas)
```

Comment:
There should be no risk for collisions on principle, i.e. some sort of linear probing must be implemented. Otherwise it creates an unfair advantage over entries which do that.

Reply:
Makes sense, I have committed a hash collision double check on both the getOrCreate and the merge functions for the map. Added about 300ms to the total execution. now at ~2.54s `Time: 2528.6563ms`, `Time: 2524.852ms` and `Time: 2596.0866ms`